source/libraries/Configuration.ts

```diff
@@ -0,0 +1,24 @@
+import * as getPort from "get-port";
+import BN = require('bn.js');
+
+export class Configuration {
+    public readonly httpProviderHost: string;
+    public readonly httpProviderPort: number;
+    public readonly gasPrice: BN;
+    public readonly privateKey: string;
+
+    public constructor(host: string, port: number, gasPrice: BN, privateKey: string) {
+        this.httpProviderHost = host;
+        this.httpProviderPort = port;
+        this.gasPrice = gasPrice;
+        this.privateKey = privateKey;
+    }
+
+    public static create = async (): Promise<Configuration> => {
```

Comment:
Is this the syntax that has to be used for static async functions?
(As opposed to `public static async create(): Promise<Configuration> {`)?

Reply:
This style makes it so the function has a captured `this` so you don't have to worry about JavaScript stupid this re-binding problems.  In general, I prefer this style over `public foo(): Type {...}` for that reason.